src/pyodide/python-entrypoint-helper.js

```diff
@@ -124,7 +124,10 @@ export default {
   async fetch(request, env, ctx) {
     try {
       const { mainModule } = await getPyodide(ctx);
-      return await mainModule.fetch.callRelaxed(request, env, ctx);
+      if (mainModule.fetch) {
+        throw new Error("Python Worker should define an on_fetch method");
+      }
```

Comment:
Definitely don't do this. If they do:
```
from js import fetch

def on_fetch():
   fetch(...)
```
it will throw an error.

Reply:
hmmm. That's unfortunate, is there a more reliable way to detect this?

Reply:
What I'll do is show this error if `on_fetch` also isn't defined. I just want to make sure we get a nice error message for this case.

Reply:
It seems a little strange that there's a slight mismatch between what the error is saying vs. what the if statement is checking. 

I would have expected something more like if (no on_fetch) throw error("No on_fetch")

Reply:
What if the user does:
```py
from js import fetch

async def on_alarm():
   resp = await fetch("http://example.com")
```
Presumably they are allowed to have an alarm handler but no fetch handler and they are allowed to fetch inside their alarm handler? Would they even see this in a clear way? Maybe wrangler should warn in the case that there is a `fetch` but no `on_fetch`, but I don't think it should be a hard error, and I don't think workerd should check for it.

Reply:
@garrettgu10 yeah, you're totally right, I should just be throwing a nicer error when `on_fetch` isn't there

Reply:
But people aren't required to have a fetch handler, right? What if the worker is just a cron job?

Reply:
Yeah, but I will only throw it in this function (which won't be called unless a `fetch` handler is needed)

Reply:
Ah so this is just trapping bugs in wrangler then?

Reply:
It's just so folks who are using the old syntax get a nice error message.